src/java.base/share/classes/java/lang/StringConcatHelper.java

```diff
@@ -432,7 +432,7 @@ static String simpleConcat(Object first, Object second) {
     @ForceInline
     static String doConcat(String s1, String s2) {
         byte coder = (byte) (s1.coder() | s2.coder());
-        int newLength = (s1.length() + s2.length()) << coder;
+        int newLength = checkOverflow(s1.length() + s2.length()) << coder;
```

Comment:
Might be mildly inefficient for this case since `checkOverflow` is designed for the compound length+coder `long`. Since this is only used in the simple path it should be easy for JITs to optimize, though.

Reply:
Exactly. The one used here is the (int) variant.

Reply:
D'oh, missed that it's been added since it's not co-located in the source. Sorry for the noise!